MULTIPLATFORM.md

```diff
@@ -0,0 +1,325 @@
+
+In this tutorial, the multiplatform application will include three parts:
+
+ * An **Android application** represented by a separate Android Studio project written in pure Kotlin
+ * An **iOS application** represented by a separate XCode project, written in Swift and using Kotlin
+ * A **multiplatform library** containing a business logic of the application.
+   This library can contain both platform-dependent and platform-independent code and is compiled into a `jar`-library
+   for Android and in a `Framework` for iOS by Gradle.
+
+In its turn, the multiplatform library will include three subprojects:
+
+ * `common` - contains a common logic for both applications;
+ * `ios` - contains an iOS-specific code;
+ * `android` - contains an Android-specific code.
+
+### 1. Preparing a workspace
+
+Let's represent the structure described above as a directory tree. Assume that our multiplatform library is intended to
+generate different greetings on different platform. Create the following directory structure:
+
+    application/
+    ├── androidApp/
+    ├── iosApp/
+    └── greeting/
+        ├── common/
+        ├── android/
+        └── ios/
+
+As said above, [Gradle](https://gradle.org/) is the main build system for Kotlin thus our project will use it.
+
+To install Gradle refer to [this instruction](https://gradle.org/install/). Despite the fact that you can use the local
+Gradle installation for building the project, it's recommended to use the
+[Gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html) instead. To create the wrapper, install
+Gradle as described above and execute `gradle wrapper` in the root directory of the project. After that you can
+use `./gradlew` to run the build instead of using your local Gradle installation.
+
+Once the wrapper is created we need to describe the project structure in Gradle terms. To do this, create
+a `settings.gradle` file in the root directory of the project and put the following snippet into it:
+
+    include ':greeting'
+    include ':greeting:common'
+    include ':greeting:android'
+    include ':greeting:ios'
+
+Here we declare all subprojects for our `greeting` multiplatform library. All other multiplatform libraries included
+in the project also must be declared here.
+
+Note that both iOS and Android applications are not included in the root Gradle build. They are represented by
+independent builds which are managed by specific IDEs (Android Studio and XCode). Such an approach makes work with
+these builds easier from these IDEs.
+
+As for IDE for other parts of the project, IntelliJ IDEA is recommended to be used.
+
+> Note: Kotlin/Native is not supported by IntelliJ IDEA so the only IDE to develop Kotlin/Native subprojects is
+CLion. But at the moment CLion has no Gradle integration. As a workaround you can create a CLion Cmake project
+from a Kotlin/Native Gradle one. Just run the `generateCMake` task for this project.
```

Comment:
If people are not familiar with gradle they don't know how to run that task.
I would recommend to say 
> Just run `./gradlew generateCMake` for this project. It will generate all the necessary files which are required.

(or smth like that :))

Reply:
Done